### DISCUSSION TITLE
Am I missing something?....Agents cannot complete a basic task without hallucination/off-course

### DISCUSSION THREAD

**Discussion by TheRichCarey** [reworkd/AgentGPT#822] [Q&A]
I've got it working easily but all this is good for is chewing through credits. Its very good at looking like it knows what its doing (and a lot of its syntax and knowledge is good) but it just goes off doing its own thing with no direction until it **always** quits? hallucination central!

I set it a goal to make a single HTML page with no styling or scripting that says hello world.

It did that in 2 steps and then spent 5 minutes iterating over the page and adding script to alert it something was clicked, importing a stylesheet it made up etc.

Even AutoGPT got this stuff right and knew when a task was "done", or at least asked for feedback.

Given it doesnt even generate the output, whats the point of this other than a cool demo? It seems like it has way too many hallucinations

I would even argue it purposely adds loads of tasks until it reaches it cap as it doesn't know how to stop?!

**Comment by jasangill1** [comment]
Your feedback is valuable, and we appreciate you sharing your experience. Our team will take it into account as we continue refining the system to deliver better results. If you have any more concerns or need further assistance, please don't hesitate to let me know! 😊

Regarding the issue you mentioned, I'd like to gather some more information. Can you please let me know if these hallucinations were happening every time you asked Agent GPT to perform a task, or was it intermittent?

🔎 It's important to note that we are an open-source project, and unlike AutoGPT, we have both a local and online version available. This means we have certain checks and balances in place to ensure the satisfaction of both versions. However, we strive to maintain consistency and improve the overall user experience across all platforms.

I hope this clarifies the context. If you have any specific examples or additional details about the hallucinations you encountered, please feel free to share them. This will help us in our ongoing efforts to enhance the system. Also, note that this reply will be posted on GitHub, so you can use GitHub syntax to add more styling if you'd like. ✨

**Comment by Heimdallr01** [comment]
I have to agree. Sometimes simple tasks are repeated in almost the same way like AgentGPT doesn't realise it has already done this exact same task. This happens pretty much everytime I'm using it. I'm using the Website.

**Comment by jasangill1** [comment]
Hello @Heimdallr01 what are your uses for it? Are the prompts simple or concise with context, Would you mind sharing prompts that you see this happen most often?

Looking forward to hearing back 😃 !!

**Reply by Heimdallr01** [reply]
This happens with simple and complex prompts. 
Just now I used this prompt:

> Conduct an extensive research on the effectiveness and efficacy of Bionic Reading. Utilize current papers and research studies for this purpose.

It used GPT4 and the max Loop number and basically just figured out it needs to do extensive research and to search for scientifically backed information to finish the task.

Using a more precise prompt:

> Conduct a precise and comprehensive research on the effectiveness and efficacy of Bionic Reading. Utilize only current papers and research studies as sources and avoid any hallucinations or the inclusion of unsubstantiated content. Stay strictly focused on the main objective and do not deviate from it, ensuring that the results include solely relevant and scientifically supported information. Your research should solely concentrate on the effectiveness and efficacy of Bionic Reading and should not be distracted by other or partial tasks. The accuracy and precision of your work are crucial to ensure a comprehensive understanding of the subject matter.

It did a Google search 2-3 times, but still focused on HOW to solve the task, not solving the actual task.
